renderdoc/driver/d3d12/d3d12_command_list_wrap.cpp

```diff
@@ -3536,14 +3536,15 @@ void WrappedID3D12GraphicsCommandList::ResetAndRecordExecuteIndirectStates(
     else
     {
       state.indirectState.argsToProcess = argumentsReplayed % numArgsPerExec + numArgsPerExec;
-      BytesToRead += comSig->sig.ByteStride;
-    }
+      if(argumentsReplayed % numArgsPerExec != 0)
+        BytesToRead += comSig->sig.ByteStride;
 
-    // skip all but the last executes we care about
-    while(argumentsReplayed > state.indirectState.argsToProcess)
-    {
-      ArgumentBufferOffset += comSig->sig.ByteStride;
-      argumentsReplayed -= numArgsPerExec;
+      // skip all but the last executes we care about
+      while(argumentsReplayed > state.indirectState.argsToProcess)
```

Comment:
the while loop can only happen if argumentsReplayed > numArgsPerExec, move it into the else clause